packages/functional-tests/pages/products/index.ts

```diff
@@ -25,16 +27,25 @@ export class SubscribePage extends BaseLayout {
   }
 
   async clickPayNow() {
-    await this.page.click('button[type=submit]');
+    const pay = this.page.locator('[data-testid="submit"]');
+    await pay.waitFor();
```

Comment:
Q: does this need a `{state: 'attached'}` as well?

Reply:
Not necessarily. Sometimes when the selector is hidden or not active, we use state: attached for it to be visible and active to be able to perform action upon. But this selector looked like it didn't need a state and was visible. But in case in the future this behaves like a flaky selector we can come back and add the state.